gef.py

```diff
@@ -2172,7 +2172,12 @@ def read_cstring_from_memory(address, max_length=GEF_MAX_STRING_LENGTH, encoding
 
     char_t = cached_lookup_type("char")
     char_ptr = char_t.pointer()
-    res = gdb.Value(address).cast(char_ptr).string(encoding=encoding).strip()
+    try:
+        res = gdb.Value(address).cast(char_ptr).string(encoding=encoding).strip()
+    except gdb.error:
+        length = min(address|(DEFAULT_PAGE_SIZE-1), max_length+1)
+        mem = read_memory(address, length).decode('utf-8')
```

Comment:
with python 2 I'm getting `[!] Command 'registers' failed to execute properly, reason: 'buffer' object has no attribute 'decode'` for this line. This seems to work though:
```python
mem = bytes(read_memory(address, length)).decode("utf-8")
```